suspendmanager.lua

```diff
@@ -146,6 +150,17 @@ local ERASABLE_DESIGNATION = {
     [df.job_type.DetailFloor]=true,
 }
 
+--- Job types that impact suspendmanager
+local FILTER_JOB_TYPES = {
+    [df.job_type.CarveTrack]=true,
+    [df.job_type.ConstructBuilding]=true,
+    [df.job_type.DestroyBuilding]=true,
+    [df.job_type.DetailFloor]=true,
+    [df.job_type.Dig]=true,
+    [df.job_type.DigChannel]=true,
+    [df.job_type.SmoothFloor]=true,
+}
```

Comment:
this is absolutely fine to write it this way, but just so you're aware, many other tools take advantage of `utils.invert` like this:
```lua
local FILTER_JOB_TYPES = utils.invert{
    df.job_type.CarveTrack,
    df.job_type.ConstructBuilding,
    df.job_type.DestroyBuilding,
    df.job_type.DetailFloor,
    df.job_type.Dig,
    df.job_type.DigChannel,
    df.job_type.SmoothFloor,
}
```

on another note, I'm not sure I understand this list. Are we looking for the list of job types that impact pathability? Looking through the header file, I came up with this:
```c
        Dig,
        CarveUpwardStaircase,
        CarveDownwardStaircase,
        CarveUpDownStaircase,
        CarveRamp,
        DigChannel,
        FellTree,
        RemoveConstruction,
        ConstructBuilding,
        RemoveStairs,
```

Reply:
ooooh I never quite understood this `utils.invert`, now I get it!

And for the purpose of this list, yes, that would be all the job impacting the access to jobs, essentially to filter out things like Urist finished his nap. I've added those.